File: VB/MasterExpandButtonDisabling/Converter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Data
Imports System.Globalization
Imports System.Windows.Media

Namespace MasterExpandButtonDisabling
	Public Class Converter
		Implements IValueConverter
		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
			If parameter IsNot Nothing Then
				Dim param As String = parameter.ToString()
				If TypeOf value Is Task AndAlso param.Equals("BoolConversion") Then
					Dim rw As Task = CType(value, Task)
					Return If(rw.Steps.Count > 0, True, False)
				End If
			End If
			If TypeOf value Is Task Then
				Dim rw As Task = CType(value, Task)
				Return If(rw.Steps.Count > 0, New SolidColorBrush(Colors.DarkBlue), New SolidColorBrush(Colors.Gray))
			End If
			Return value
		End Function

		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
			Return value
		End Function
	End Class
End Namespace

File: CS/MasterExpandButtonDisabling/Converter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Data;
using System.Globalization;
using System.Windows.Media;

namespace MasterExpandButtonDisabling {
    public class Converter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            if (parameter != null) {
                string param = parameter.ToString();
                if (value is Task && param.Equals("BoolConversion")) {
                    Task rw = (Task)value;
                    return rw.Steps.Count > 0 ? true : false;
                }
            }
            if (value is Task) {
                Task rw = (Task)value;
                return rw.Steps.Count > 0 ? new SolidColorBrush(Colors.DarkBlue) : new SolidColorBrush(Colors.Gray);
            }
            return value;
        }

        public object ConvertBack(object value, Type targetType,
            object parameter, CultureInfo culture) {
            return value;
        }
    }
}
